src/DIRAC/DataManagementSystem/Service/FileCatalogHandler.py

```diff
@@ -261,6 +261,12 @@ def export_getFileMetadata(self, lfns):
         """Get the metadata associated to supplied lfns"""
         return self.fileCatalogDB.getFileMetadata(lfns, self.getRemoteCredentials())
 
+    types_getFileDetails = [[list, dict, str]]
+
+    def export_getFileDetails(self, lfns):
```

Comment:
Out of consistency, would you call this 
```suggestion
    def export_getFileDetailsPublic(self, lfns):
```

But this sounds better:

```suggestion
    def export_getPublicFileDetails(self, lfns):
```

Reply:
Well, depends of the consistency with what.

I went for consistency with `export_getFileMetadata`, and the `Public` is supposed to point out that that one has the permission check, while the non public one doesn't have the permission check.
And the file details aren't public, I think? I don't know who can see which meta data to be honest.

Maybe getFileUserMetadata would be better anyway, because `Details` is very vague?

Reply:
I meant consistency with the method name used at DB level.

Reply:
But the exported function should be clearer for the user, and not worry about the implementation details underneath. So, `export_getFileUserMetadata` might be better?

Reply:
Fine for me!

Reply:
Ahh, that function name already exists, but is not working for bulk :/

Reply:
@andresailer if you sort this one out I can merge it this afternoon (and make a release with it just after).

Reply:
Just use a different name that doesn't exist yet, or something more involved?

Reply:
Just use a different name

Reply:
Went back to `export_getFileDetails`